docs/usage.rst

```diff
@@ -0,0 +1,197 @@
+
+Command line syntax for Naggum Compiler is::
+
+    $ Naggum.Compiler [source1.naggum] [source2.naggum] [/r:assembly1] [/r:assembly2]
+
+Each input source file will be compiled to a separate executable assembly (i.e.
+an ``.exe`` file) in the current directory.
+
+``.naggum`` extension is recommended for high level Naggum files.
+
+Naggum Assembler
+----------------
+
+Naggum Assembler uses low level Naggum dialect. Command line syntax is::
+
+    $ Naggum.Assembler [source1.nga] [source2.nga]
+
+Assembler output is determined based on input file content: each file may
+contain multiple (or none) assembly constructs. Each assembly will be saved to
+its own executable file in the current directory.
+
+``.nga`` extension is recommended for low level Naggum files.
+
+S-expression syntax
+-------------------
+
+Each Naggum program (either high-level or low-level) is written as a sequence of
+S-expression forms. In s-expression, everything is either an atom or a list.
+Atoms are written as-is, lists should be taken into parens.
+
+Possible atom values are::
+
+    "A string"
+    1.4e-5 ; a number
+    System.Console ; a symbol
+
+A symbol may consist of any letter, digit or ``+-*/=<>!?.`` character sequence.
+
+Lists are simply sequences of s-expression in parens::
+
+    (this is a list)
+
+    (this (is ("Also") a.list))
+
+Naggum source code may also include comments. Everything after ``;`` character
+will be ignored till the end of the line::
+
+    (valid atom) ; this is a comment
+
+Low level syntax
+----------------
+
+Naggum low level syntax is closer to `CIL`_. It may be used to define CLI
+constructs such as assemblies, modules, types and methods. Every ``.nga`` file
+may contain multiple (including zero) assembly definitions.
+
+Assembly definition
+^^^^^^^^^^^^^^^^^^^
+
+Assembly defitinion should have the following form::
+
+    (.assembly Name
+        Item1
+        Item2
+        ...)
+
+Assembly items can be methods and types. Top level methods defined in an
+``.assembly`` form will be compiled to global CIL functions.
+
+Currently there is no support for type definition inside an assembly.
+
+Each assembly may contain one entry point method (either a static type method or
+an assembly global function marked by ``.entrypoint`` property).
+
+Method definition
+^^^^^^^^^^^^^^^^^
+
+Method definition should have the following form::
+
+    (.method Name (argument types) return-type (metadata items)
+        body-statements
+        ...)
+
+Method argument and return types should be fully-qualified (e.g. must include a
+namespace: for example, ``System.Void``).
+
+The only supported metadata item is ``.entrypoint``. It marks a method as an
+assembly entry point.
+
+Method example::
+
+    (.method Main () System.Void (.entrypoint)
+        (ldstr "Hello, world!")
+        (call (mscorlib System.Console WriteLine (System.String) System.Void))
+        (ret))
+
+Method body should be a CIL instruction sequence.
+
+CIL instructions
```

Comment:
Up till now, all titles were in singular form. I'd use plural everywhere, but it's up to you.